cpu/sam3/cpu.c

```diff
@@ -56,6 +56,9 @@ void cpu_init(void)
     /* unlock write protect register for PMC module */
     PMC->PMC_WPMR = PMC_WPMR_WPKEY(WPKEY);
 
+    /* enable Synchronous Data Output for PIOB */
+    PIOB->PIO_OWER = ~0l;
```

Comment:
I would rather not enable write on all pins. As you pointed out already, one could either limit this to the specific pin for LED0 (PB27) *or* use set/clear calls instead. The problem with using `PIO_OWER` is that is has to be set somewhere else not related to the toggle operation, so if one doesn't use LED0 its still set. 

I personally would rather go for something like:

```
#define LED0_TOGGLE         ((PIOB->PIO_ODSR) ?  LED0_OFF : LED0_ON)
```

if that is possible and works - I don't have a sam3 based boards for testing 😞

Reply:
This is the correct macro for boards/arduino-due/include/board.h:
`#define LED0_TOGGLE         (((PIOB->PIO_ODSR & PIO_PB27) == PIO_PB27) ?  LED0_OFF : LED0_ON)`
And this only enables LED0 in cpu/sam3/cpu.c:
`PIOB->PIO_OWER |=  PIO_PB27;`
I've tested both solutions and they work.
Which one? =)

Reply:
ah right, missed the check for the specific GPIO - however shouldn't 

```
#define LED0_TOGGLE ((PIOB->PIO_ODSR & PIO_PB27) ? LED0_OFF : LED0_ON)
```

be shorter and sufficient, too? And then I would prefer this variant, because it does not require to change `PIOB->PIO_OWER = ~0l;` somewhere unrelated with the LED macro.

Reply:
My bad, you're right, that check was unneeded.
I also think changing the macro is the cleanest solution.